wasm/jit/jit_arm64.go

```diff
@@ -1773,36 +1780,211 @@ func (c *arm64Compiler) compileGe(o *wazeroir.OperationGe) error {
 }
 
+// compileLoad32 implements compiler.compileLoad32 for the amd64 architecture.
 func (c *arm64Compiler) compileLoad32(o *wazeroir.OperationLoad32) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	var loadInst obj.As
+	if o.Signed {
+		loadInst = arm64.AMOVW
+	} else {
+		loadInst = arm64.AMOVWU
+	}
+	return c.compileLoadImpl(o.Arg.Offset, loadInst, 32/8, false)
 }
 
+// compileLoadImpl implements compileLoadImpl* variants for arm64 architecture.
+func (c *arm64Compiler) compileLoadImpl(offsetArg uint32, loadInst obj.As, targetSizeInBytes int64, isFloat bool) error {
+	offsetReg, err := c.compileMemoryAccessOffsetSetup(offsetArg, targetSizeInBytes)
+	if err != nil {
+		return err
+	}
+
+	resultRegister := offsetReg
+	if isFloat {
+		resultRegister, err = c.allocateRegister(generalPurposeRegisterTypeFloat)
+		if err != nil {
+			return err
+		}
+	}
+
+	// "resultRegister = [reservedRegisterForMemory + offsetReg]"
+	// In other words, "resultRegister = memory.Buffer[offset: offset+targetSizeInBytes]"
+	c.compileMemoryWithRegisterOffsetToRegisterInstruction(
+		loadInst,
+		reservedRegisterForMemory, offsetReg,
+		resultRegister,
+	)
+
+	c.locationStack.pushValueLocationOnRegister(resultRegister)
+	return nil
+}
+
+// compileStore implements compiler.compileStore for the amd64 architecture.
 func (c *arm64Compiler) compileStore(o *wazeroir.OperationStore) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	var movInst obj.As
+	var targetSizeInBytes int64
+	switch o.Type {
+	case wazeroir.UnsignedTypeI32:
+		movInst = arm64.AMOVW
+		targetSizeInBytes = 32 / 8
+	case wazeroir.UnsignedTypeI64:
+		movInst = arm64.AMOVD
+		targetSizeInBytes = 64 / 8
+	case wazeroir.UnsignedTypeF32:
+		movInst = arm64.AFMOVS
+		targetSizeInBytes = 32 / 8
+	case wazeroir.UnsignedTypeF64:
+		movInst = arm64.AFMOVD
+		targetSizeInBytes = 64 / 8
+	}
+	return c.compileStoreImpl(o.Arg.Offset, movInst, targetSizeInBytes)
 }
 
+// compileStore8 implements compiler.compileStore8 for the amd64 architecture.
 func (c *arm64Compiler) compileStore8(o *wazeroir.OperationStore8) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	return c.compileStoreImpl(o.Arg.Offset, arm64.AMOVB, 1)
 }
 
+// compileStore16 implements compiler.compileStore16 for the amd64 architecture.
 func (c *arm64Compiler) compileStore16(o *wazeroir.OperationStore16) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	return c.compileStoreImpl(o.Arg.Offset, arm64.AMOVH, 16/8)
 }
 
+// compileStore32 implements compiler.compileStore32 for the amd64 architecture.
 func (c *arm64Compiler) compileStore32(o *wazeroir.OperationStore32) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	return c.compileStoreImpl(o.Arg.Offset, arm64.AMOVW, 32/8)
+}
+
+// compileStoreImpl implements compleStore* variants for arm64 architecture.
+func (c *arm64Compiler) compileStoreImpl(offsetArg uint32, storeInst obj.As, targetSizeInBytes int64) error {
```

Comment:
is int64 excessive
```suggestion
func (c *arm64Compiler) compileStoreImpl(offsetArg uint32, storeInst obj.As, targetSizeInBytes uint32) error {
```

Reply:
agreed, but in anyway we end up casting this to int64 to pass to assembler, so I intentionally use int64 here. I will leave as-is! Thanks!